src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -3620,6 +3620,12 @@ public override async Task<RachisConnection> ConnectToPeer(string url, string ta
                     stream = result.Stream;
                     supportedFeatures = result.SupportedFeatures;
 
+                    if (supportedFeatures.DataCompression)
+                    {
+                        stream = new ReadWriteCompressedStream(stream);
```

Comment:
this is when you create the connection, where is this handled on the receiving end?

Reply:
https://github.com/ravendb/ravendb/blob/2b7dd28d2e3a0a4b2d2bfb5b8bd2143ddbc10743/src/Raven.Server/RavenServer.cs#L1958

```
                            if (ShouldUseDataCompression(header))
                            {
                                stream = new ReadWriteCompressedStream(stream, buffer);
                                tcp.Stream = stream;
                            }
```
look at the changes on the ```ShouldUseDataCompression``` method.